File: API/Entities/Product.cs
﻿namespace API.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public int Price { get; set; }
        public string Type { get; set; } = string.Empty;
        public string Brand { get; set; } = string.Empty;
        public string PictureUrl { get; set; } = string.Empty;
        public int QuantityInStock { get; set; }

    }
}

File: UI/src/models/product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  type?: string;
  brand: string;
  pictureUrl: string;
  quantityInStock?: number;
}

export interface ProductParams {
  orderBy: string;
  searchTerm?: string;
  brands: string[];
  types: string[];
  pageNumber: number;
  pageSize: number;
}
